core/src/main/java/feast/core/job/InMemoryJobRepository.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.job;
+
+import com.google.common.collect.Lists;
+import feast.common.models.FeatureSetReference;
+import feast.core.model.Job;
+import feast.core.model.JobStatus;
+import feast.proto.core.SourceProto;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Keeps state of all jobs managed by current application in memory. On start loads persistent state
+ * through JobManager from JobManager backend.
```

Comment:
I think we should add an example here like `e.g., using Dataflow to maintain state about jobs` so that it is less abstract.